ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpConnection.java

```diff
@@ -0,0 +1,23 @@
+package io.ripc.protocol.tcp;
+
+import org.reactivestreams.Publisher;
+
+/**
+ * An abstraction for a TCP connection.
+ *
+ * @param <R> The type of objects read from this connection.
+ * @param <W> The type of objects written to this connection.
+ */
+public interface TcpConnection<R, W> extends Publisher<R> {
```

Comment:
> Why having a separate tcp and ripc-core

Have a look at issue #18 where we are suggesting a different project structure.
ripc-core would have protocol neutral classes and different protocols will be different projects.

> Looking at the artifacts there is nothing specific yet to Tcp

The implementations are TCP specific. Connection concept is protocol specific (connection based protocols). UDP would not have a connection as an example.

> Same for Server and Client, does it have to be in separate module?

There is no client as of now. Yeah, they should be in a single module/project per protocol.

> Even in a case of File streaming I could see the same abstraction working as well.

You mean file streaming over TCP, HTTP?